force-app/main/default/lwc/newServiceSchedule/newServiceSchedule.js

```diff
@@ -25,131 +8,126 @@ const LARGE_SIZE = 12;
-        dateTime: DATE_TIME_LABEL,
-    };
-
+    objectApiName;
+    labels;
     sizes = {
         large: LARGE_SIZE,
         small: SMALL_SIZE,
     };
 
-    dateFields = {
-        start: FIRST_SESSION_START_FIELD.fieldApiName,
-        end: FIRST_SESSION_END_FIELD.fieldApiName,
-        seriesEndsOn: END_DATE_FIELD.fieldApiName,
-        numberOfSessions: NUMBER_OF_SESSIONS_FIELD.fieldApiName,
-    };
+    _serviceScheduleModel;
+    @track
+    picklistFields;
+    requiredFields;
+    dateFields;
+    fieldSet;
+    isLoaded = false;
 
-    picklistFields = {
-        frequency: FREQUENCY_FIELD.fieldApiName,
-        daysOfWeek: DAYS_OF_WEEK_FIELD.fieldApiName,
-        seriesEnds: SERVICE_SECHEDULE_ENDS_FIELD.fieldApiName,
-    };
+    @api
+    get serviceScheduleModel() {
+        return this._serviceScheduleModel;
+    }
+
+    set serviceScheduleModel(value) {
+        this.isLoaded = false;
+        // This is a nested object so the inner objects are still read only
+        this._serviceScheduleModel = JSON.parse(JSON.stringify(value));
+
+        this.labels = this._serviceScheduleModel.labels.serviceSchedule;
+        this.objectApiName = this._serviceScheduleModel.labels.serviceSchedule.objectApiName;
+        this.requiredFields = this._serviceScheduleModel.scheduleRequiredFields;
+        this.dateFields = this._serviceScheduleModel.scheduleRecurrenceDateFields;
+        this.picklistFields = this._serviceScheduleModel.scheduleRecurrencePicklistFields;
+        this.fieldSet = this._serviceScheduleModel.scheduleInformationFields;
 
-    @wire(getFieldSet, {
-        objectName: SERVICE_SCHEDULE_OBJECT.objectApiName,
-        fieldSetName: FIELD_SET_NAME,
-    })
-    wiredFieldSet({ error, data }) {
-        if (data) {
-            this.setFieldSet(data);
-        } else if (error) {
-            console.log(JSON.stringify(error));
-        }
+        this.processFields();
+        this.isLoaded = true;
     }
 
-    setFieldSet(data) {
-        // TODO: Test if this eliminates duplicate fields
-        let recurrenceFields = Object.values(this.dateFields);
-        recurrenceFields.push(Object.values(this.picklistFields));
-
-        this.fieldSet = data
-            .filter(obj => !recurrenceFields.includes(obj.apiName))
-            .map(obj => {
-                let field = { ...obj };
-                field.size =
-                    obj.apiName === NAME_FIELD.fieldApiName ? LARGE_SIZE : SMALL_SIZE;
+    @api reportValidity() {
+        return [...this.template.querySelectorAll("lightning-input-field")].reduce(
+            (validSoFar, inputField) => {
+                return validSoFar && inputField.reportValidity();
+            },
+            true
+        );
+    }
+
+    @api
+    get serviceSchedule() {
+        let serviceSchedule = this._serviceScheduleModel.serviceSchedule;
+
+        [...this.template.querySelectorAll("lightning-input-field")].forEach(field => {
```

Comment:
To turn it into an array, querySelectorAll returns... "A non-live NodeList containing one Element object for each element that matches at least one of the specified selectors or an empty NodeList in case of no matches." In order to iterate you have to convert it to an array although I prolly don't need to spread it here. I will test it out.

Reply:
Tested it out and I do not get any errors without copying the array this way but it doesn't grab the values. I'm not exactly sure why though.

Reply:
Cool, thanks. Let's leave it as-is, then.